OpenRA.Mods.Common/Traits/Sound/VoiceAnnouncement.cs

```diff
@@ -44,6 +47,9 @@ protected override void TraitEnabled(Actor self)
 			if (IsTraitDisabled)
 				return;
 
+			if (!Info.AnnounceWhenSpawnedByMap && self.World.WorldTick == 0)
```

Comment:
Checking against WorldTick is likely to cause issues in the future, so I would prefer we did this using conditions instead.